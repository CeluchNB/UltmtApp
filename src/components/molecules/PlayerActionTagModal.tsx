import BaseModal from '../atoms/BaseModal'
import { Chip } from 'react-native-paper'
import PrimaryButton from '../atoms/PrimaryButton'
import React from 'react'
import ScreenTitle from '../atoms/ScreenTitle'
import SecondaryButton from '../atoms/SecondaryButton'
import UserInput from '../atoms/UserInput'
import { useColors } from '../../hooks'
import { FlatList, LogBox, StyleSheet, View } from 'react-native'
import {
    addTag,
    selectTags,
} from '../../store/reducers/features/game/liveGameReducer'
import { useDispatch, useSelector } from 'react-redux'

interface PlayerActionTagModalProps {
    visible: boolean
    onClose: (submit: boolean, tags: string[]) => void
}

const PlayerActionTagModal: React.FC<PlayerActionTagModalProps> = ({
    visible,
    onClose,
}) => {
    LogBox.ignoreLogs(['`flexWrap: `wrap`` is'])
    const { colors } = useColors()
    const dispatch = useDispatch()
    const tags = useSelector(selectTags)
    const [newTag, setNewTag] = React.useState('')
    const [selectedTags, setSelectedTags] = React.useState<string[]>([])

    const onAddTag = () => {
        if (newTag.length > 0) {
            dispatch(addTag(newTag))
        }
        setNewTag('')
    }

    const toggleTagSelection = (item: string) => {
        if (selectedTags.includes(item)) {
            setSelectedTags(prev => {
                return prev.filter(s => s !== item)
            })
        } else {
            setSelectedTags(prev => {
                return [item, ...prev]
            })
        }
    }

    const onChangeText = (text: string) => {
        setNewTag(text)
    }

    const styles = StyleSheet.create({
        chip: {
            borderRadius: 8,
            margin: 5,
            backgroundColor: colors.primary,
            borderWidth: 2,
        },
        listContainer: {
            height: 200,
        },
        flatListContainer: {
            flexDirection: 'row',
            flexWrap: 'wrap',
        },
        newTagContainer: {
            flexDirection: 'row',
            marginBottom: 10,
        },
        input: {
            flex: 1,
            marginRight: 10,
        },
        addButtonContainer: {
            alignSelf: 'center',
        },
    })

    return (
        <BaseModal
            visible={visible}
            onClose={() => {
                onClose(false, [])
            }}>
            <ScreenTitle title="Tags" />
            <View style={styles.listContainer}>
                <FlatList
                    contentContainerStyle={styles.flatListContainer}
                    scrollEnabled={true}
                    scrollToOverflowEnabled={true}
                    data={tags}
                    renderItem={({ item }) => {
                        return (
                            <Chip
                                style={styles.chip}
                                mode="outlined"
                                onPress={() => {
                                    toggleTagSelection(item)
                                }}
                                selectedColor={
                                    selectedTags.includes(item)
                                        ? colors.textPrimary
                                        : colors.gray
                                }
                                ellipsizeMode="tail">
                                {item}
                            </Chip>
                        )
                    }}
                />
            </View>
            <View style={styles.newTagContainer}>
                <UserInput
                    style={styles.input}
                    placeholder="New tag..."
                    value={newTag}
                    onChangeText={onChangeText}
                />
                <View style={styles.addButtonContainer}>
                    <SecondaryButton
                        text="add"
                        onPress={async () => {
                            onAddTag()
                        }}
                    />
                </View>
            </View>
            <PrimaryButton
                text="done"
                loading={false}
                onPress={async () => {
                    onClose(true, selectedTags)
                    setSelectedTags([])
                }}
            />
        </BaseModal>
    )
}

export default PlayerActionTagModal
