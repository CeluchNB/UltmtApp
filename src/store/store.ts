import accountReducer from './reducers/features/account/accountReducer'
import { configureStore } from '@reduxjs/toolkit'
import liveGameReducer from './reducers/features/game/liveGameReducer'
import livePointReducer from './reducers/features/point/livePointReducer'
import managedTeamReducer from './reducers/features/team/managedTeamReducer'

const store = configureStore({
    reducer: {
        account: accountReducer,
        managedTeam: managedTeamReducer,
        liveGame: liveGameReducer,
        livePoint: livePointReducer,
    },
})

export type RootState = ReturnType<typeof store.getState>
export default store
