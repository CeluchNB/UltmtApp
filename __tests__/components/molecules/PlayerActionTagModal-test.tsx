import PlayerActionTagModal from '../../../src/components/molecules/PlayerActionTagModal'
import { Provider } from 'react-redux'
import React from 'react'
import store from '../../../src/store/store'
import { act, fireEvent, render, waitFor } from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
const originalWarn = console.warn.bind(console.warn)
beforeAll(() => {
    console.warn = msg =>
        !msg.toString().includes('flexWrap: `wrap`') && originalWarn(msg)
})
afterAll(() => {
    console.warn = originalWarn
})

it('should match snapshot', () => {
    const snapshot = render(
        <Provider store={store}>
            <PlayerActionTagModal visible={true} onClose={jest.fn()} />
        </Provider>,
    )

    expect(snapshot.toJSON()).toMatchSnapshot()
})

it('should add tag correctly', async () => {
    const { getByPlaceholderText, getByText } = render(
        <Provider store={store}>
            <PlayerActionTagModal visible={true} onClose={jest.fn()} />
        </Provider>,
    )

    const tagInput = getByPlaceholderText('New tag...')
    fireEvent.changeText(tagInput, 'test')

    const addBtn = getByText('add')
    fireEvent.press(addBtn)

    await waitFor(() => {
        expect(getByText('test')).toBeTruthy()
    })
})

it('should call close correctly', async () => {
    const onClose = jest.fn()
    const { getByText } = render(
        <Provider store={store}>
            <PlayerActionTagModal visible={true} onClose={onClose} />
        </Provider>,
    )

    const huckBtn = getByText('huck')
    fireEvent.press(huckBtn)

    const doneBtn = getByText('done')
    fireEvent.press(doneBtn)
    await act(async () => {})

    expect(onClose).toBeCalledWith(['huck'])
})
