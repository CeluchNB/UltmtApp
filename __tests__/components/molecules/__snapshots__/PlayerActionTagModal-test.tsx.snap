// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  testID="base-modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#30303050",
        "display": "flex",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1a1a1a",
          "borderColor": "#ffd600",
          "borderRadius": 10,
          "color": "#ffd600",
          "elevation": 5,
          "flex": 1,
          "margin": 20,
          "padding": 25,
          "shadowColor": "#ffd600",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#ffd600",
              "fontSize": 61.64383561643835,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Tags
      </Text>
      <View
        style={
          {
            "height": 200,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
          data={
            [
              "huck",
              "break",
              "layout",
            ]
          }
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={50}
          scrollToOverflowEnabled={true}
          stickyHeaderIndices={[]}
          viewabilityConfigCallbackPairs={[]}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                collapsable={false}
                style={
                  {
                    "alignSelf": undefined,
                    "bottom": undefined,
                    "end": undefined,
                    "left": undefined,
                    "position": undefined,
                    "right": undefined,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "start": undefined,
                    "top": undefined,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "backgroundColor": "#303030",
                        "borderColor": "rgba(206, 198, 180, 0.29)",
                        "borderRadius": 8,
                        "borderStyle": "solid",
                        "borderWidth": 2,
                        "flexDirection": "column",
                        "margin": 5,
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "disabled": false,
                          "selected": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          false,
                          [
                            {
                              "borderRadius": 8,
                            },
                            {
                              "flexGrow": 1,
                            },
                          ],
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "flexGrow": 1,
                              "paddingLeft": 4,
                              "position": "relative",
                            },
                            {
                              "paddingLeft": 0,
                            },
                            {
                              "paddingRight": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "rgba(28, 27, 31, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              {
                                "textAlign": "left",
                              },
                              {
                                "writingDirection": "ltr",
                              },
                              [
                                {
                                  "marginVertical": 6,
                                  "textAlignVertical": "center",
                                },
                                {
                                  "color": "#CEC6B4",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                {
                                  "marginLeft": 16,
                                  "marginRight": 16,
                                },
                                undefined,
                              ],
                            ]
                          }
                        >
                          huck
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                collapsable={false}
                style={
                  {
                    "alignSelf": undefined,
                    "bottom": undefined,
                    "end": undefined,
                    "left": undefined,
                    "position": undefined,
                    "right": undefined,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "start": undefined,
                    "top": undefined,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "backgroundColor": "#303030",
                        "borderColor": "rgba(206, 198, 180, 0.29)",
                        "borderRadius": 8,
                        "borderStyle": "solid",
                        "borderWidth": 2,
                        "flexDirection": "column",
                        "margin": 5,
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "disabled": false,
                          "selected": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          false,
                          [
                            {
                              "borderRadius": 8,
                            },
                            {
                              "flexGrow": 1,
                            },
                          ],
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "flexGrow": 1,
                              "paddingLeft": 4,
                              "position": "relative",
                            },
                            {
                              "paddingLeft": 0,
                            },
                            {
                              "paddingRight": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "rgba(28, 27, 31, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              {
                                "textAlign": "left",
                              },
                              {
                                "writingDirection": "ltr",
                              },
                              [
                                {
                                  "marginVertical": 6,
                                  "textAlignVertical": "center",
                                },
                                {
                                  "color": "#CEC6B4",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                {
                                  "marginLeft": 16,
                                  "marginRight": 16,
                                },
                                undefined,
                              ],
                            ]
                          }
                        >
                          break
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                collapsable={false}
                style={
                  {
                    "alignSelf": undefined,
                    "bottom": undefined,
                    "end": undefined,
                    "left": undefined,
                    "position": undefined,
                    "right": undefined,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "start": undefined,
                    "top": undefined,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "backgroundColor": "#303030",
                        "borderColor": "rgba(206, 198, 180, 0.29)",
                        "borderRadius": 8,
                        "borderStyle": "solid",
                        "borderWidth": 2,
                        "flexDirection": "column",
                        "margin": 5,
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityState={
                        {
                          "disabled": false,
                          "selected": false,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        [
                          {
                            "overflow": "hidden",
                          },
                          false,
                          [
                            {
                              "borderRadius": 8,
                            },
                            {
                              "flexGrow": 1,
                            },
                          ],
                        ]
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "flexGrow": 1,
                              "paddingLeft": 4,
                              "position": "relative",
                            },
                            {
                              "paddingLeft": 0,
                            },
                            {
                              "paddingRight": 0,
                            },
                          ]
                        }
                      >
                        <Text
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          selectable={false}
                          style={
                            [
                              {
                                "color": "rgba(28, 27, 31, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              {
                                "textAlign": "left",
                              },
                              {
                                "writingDirection": "ltr",
                              },
                              [
                                {
                                  "marginVertical": 6,
                                  "textAlignVertical": "center",
                                },
                                {
                                  "color": "#CEC6B4",
                                  "fontFamily": "System",
                                  "fontSize": 14,
                                  "fontWeight": "500",
                                  "letterSpacing": 0.1,
                                  "lineHeight": 20,
                                },
                                {
                                  "marginLeft": 16,
                                  "marginRight": 16,
                                },
                                undefined,
                              ],
                            ]
                          }
                        >
                          layout
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 10,
          }
        }
      >
        <View
          style={
            {
              "flex": 1,
              "marginRight": 10,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                {
                  "backgroundColor": "#303030",
                  "borderColor": "#7AC6FF",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                undefined,
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                [
                  {
                    "paddingBottom": 0,
                  },
                  {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                accessibilityLabel="New tag..."
                autoCapitalize="none"
                editable={true}
                keyboardType="default"
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="New tag..."
                placeholderTextColor="#ffffff"
                selectionColor="#ffd600"
                style={
                  [
                    {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    {
                      "height": 48,
                    },
                    {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    {
                      "color": "#ffd600",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": undefined,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    [
                      {},
                    ],
                    undefined,
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignSelf": "center",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignSelf": undefined,
                "bottom": undefined,
                "end": undefined,
                "left": undefined,
                "position": undefined,
                "right": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "start": undefined,
                "top": undefined,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderColor": "#ffd600",
                    "borderRadius": 20,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "minWidth": 64,
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      false,
                      {
                        "borderRadius": 20,
                      },
                    ]
                  }
                  testID="button"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "textAlign": "left",
                          },
                          {
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "marginHorizontal": 16,
                              "marginVertical": 9,
                              "textAlign": "center",
                            },
                            false,
                            {
                              "marginHorizontal": 24,
                              "marginVertical": 10,
                            },
                            undefined,
                            {
                              "textTransform": "uppercase",
                            },
                            {
                              "color": "#ffd600",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      add
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "alignSelf": undefined,
            "bottom": undefined,
            "end": undefined,
            "left": undefined,
            "position": undefined,
            "right": undefined,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "start": undefined,
            "top": undefined,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "#ffd600",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 0,
                "minWidth": "auto",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  false,
                  {
                    "borderRadius": 20,
                  },
                ]
              }
              testID="button"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      {
                        "textAlign": "left",
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 24,
                          "marginVertical": 10,
                        },
                        {
                          "marginHorizontal": 8,
                        },
                        {
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  done
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
